src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,69 @@
-export const Pagination = () => {};
+import React from 'react';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: React.Dispatch<React.SetStateAction<number>>;
+};
+
+export const Pagination = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}: Props) => {
+  const numOfPages = Math.ceil(total / perPage);
+  const pages = Array.from({ length: numOfPages }, (_, i) => i + 1);
+  const prevHandler = () => currentPage !== 0 && onPageChange(currentPage - 1);
```

Comment:
While this function will work correctly, it's important to note that you are creating a new function every time the component re-renders. This can lead to performance issues in larger applications. Consider using the useCallback hook to memoize this function.